portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -0,0 +1,127 @@
+import java.util.List;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/comment")
+public class CommentServlet extends HttpServlet {
+
+  /** The text param is a string property identifier for setting and retrieving
+   * the body of textual content stored for a comment made by the user. */
+  private static final String TEXT_PARAM = "text";
+
+  /** The textin param is a unique string identifier for retrieving the comment
+   * text from a user through the form input in POST requests */
+  private static final String TEXTIN_PARAM = "text-input";
+
+  /** The name param is string property identifier for setting and retrieving
+   * the name of the user associated with a submitted comment */
+  private static final String NAME_PARAM = "name";
+
+  /** The time param is a string property identifier for the time of submission 
+   * upon a POST request for a submitted comment. It is also the property used
+   * to sort comments in most recently submitted order */
+  private static final String TIME_PARAM = "timestamp";
+  
+  /** The tag param is a string property identifier for comments. It is the 
+   * property used to determine which comments are displayed with their 
+   * corresponding blog posts. If not included in a GET request, no comments
+   * will be returned. */
+  private static final String TAG_PARAM = "tag";
+
+  /** The url param is a string identifier for the page url that the associated 
+   * comments will be posted to and where the user will be redirected to after 
+   * posting a comment. If not included POST request, user will not be 
+   * redirected to the original page after submitting a comment */
+  private static final String URL_PARAM = "current-url";
+
+  /** The comment param specifices the string identifier for the entity class 
+   * that the GET request queries */
+  private static final String COMMENT = "Comment";
+
+  /** The path len param specifices the integer identifier for the length of
+   * the blog url path up until its tag specifier. If not included in the POST
+   * request, the blog post tag cannot be retrieved for subsequent association
+   * with the submitted comment */
+  private static final int PATH_LEN = 15;
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String requestTag = request.getParameter(TAG_PARAM);
+    Query.FilterPredicate filter = new Query.FilterPredicate(TAG_PARAM, FilterOperator.EQUAL, requestTag);
+    Query query = new Query(COMMENT).setFilter(filter).addSort(TIME_PARAM, SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    List<Comment> comments = new ArrayList<>();
+
+    if (requestTag != null) {
```

Comment:
Perhaps this check should be earlier? No sense in doing a DB query if it's null and you're not going to use the results.